From 8499ae7e4ff0cf798367ebe6ea9fb991aa43db6c Mon Sep 17 00:00:00 2001
From: Ed Catmur <ed@catmur.uk>
Date: Fri, 24 Nov 2023 08:53:05 -0700
Subject: [PATCH] Use static const for next/prior from C++17

Trying to form next/prior in constant evaluation may be ill-formed; see https://github.com/boostorg/mpl/issues/69
---
 include/boost/mpl/aux_/integral_wrapper.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/boost/mpl/aux_/integral_wrapper.hpp b/include/boost/mpl/aux_/integral_wrapper.hpp
index 8748fbb93a113250147e4539813990cd98916dd7..661aa307362eeb0f1ae126e7e62eb996dbba8cfa 100644
--- a/include/boost/mpl/aux_/integral_wrapper.hpp
+++ b/include/boost/mpl/aux_/integral_wrapper.hpp
@@ -56,7 +56,8 @@ struct AUX_WRAPPER_NAME
 // have to #ifdef here: some compilers don't like the 'N + 1' form (MSVC),
 // while some other don't like 'value + 1' (Borland), and some don't like
 // either
-#if BOOST_WORKAROUND(__EDG_VERSION__, <= 243)
+#if BOOST_WORKAROUND(__EDG_VERSION__, <= 243) \
+    || __cplusplus >= 201703L
  private:
     BOOST_STATIC_CONSTANT(AUX_WRAPPER_VALUE_TYPE, next_value = BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (N + 1)));
     BOOST_STATIC_CONSTANT(AUX_WRAPPER_VALUE_TYPE, prior_value = BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (N - 1)));
